src/main/java/htsjdk/samtools/sra/SRAIndexedSequenceFile.java

```diff
@@ -84,18 +78,15 @@ public ReferenceSequence getSubsequenceAt(String contig, long start, long stop)
         byte[] bases;
 
         try {
-            HashMap<Integer, Reference> localRefs = cachedReferences.get();
-            if (localRefs == null) {
-                localRefs = new HashMap<Integer, Reference>();
-                cachedReferences.set(localRefs);
+            Reference reference;
+            synchronized (this) {
+                if (cachedReference == null || !cachedReference.getCanonicalName().equals(contig)) {
```

Comment:
nothing bad will happen, just re-initialization of some data structures and re-open of file(s)